### DISCUSSION TITLE
XDrip Web Follower with new Librelinkup v 4.12

### DISCUSSION THREAD

**Discussion by HoLi1969** [NightscoutFoundation/xDrip#3794] [Q&A]
Hi, I am just a user and this issue is discussed since the last change of version in librelink. As the last 2 Nightlies do not solve the issue ... Is some of the experts changing the code for v14.2 or did I miss something? We really need xDrip back and not to bridge all again. Many many thanks for help.

**Comment by PhilTU** [comment]
Running Librelink with xDrip+ both using Bluetooth is not supported. Also which country - Librelink is regionalised.

**Reply by HoLi1969** [reply]
I speak about xDrip+ Web follower function not working since more than a week because librelinkup was changed. There is a solution described where to add new version in the code but I am not a programmer so my hope is that someone will put the change in a next nightly. This problem happened in the past, too.

**Reply by Navid200** [reply]
In that case, the title should include web follower.

**Reply by HoLi1969** [reply]
Thanks, I just changed. Sorry

**Reply by PhilTU** [reply]
ah, same as https://github.com/NightscoutFoundation/xDrip/discussions/2893

**Reply by HoLi1969** [reply]
Yes, that is the solution I found. But I have no idea how to change the conf script myself. I am just a user. I guess there are many more people without xDrip at the moment so who can help?

**Reply by PhilTU** [reply]
Navid was looking at it and I offered to test. The web follower script wasn't his thing though.

To be honest the communication of the issue has been a cluster!uck. Is it universal or do some people / regions still gave service?

**Reply by HoLi1969** [reply]
Sorry, no idea. I attach the system status screen just to be sure the topic is allocated correctly 
![Uploading Screenshot_2024-12-05-11-05-54-25_2dd2c1bcb8f659cd696bcba2072f751b.jpg…]()

**Reply by PhilTU** [reply]
I am thinking it has few users with problems, or few users at all ?

**Reply by PetBom** [reply]
I have the issue too, and it seems that more users in my network is experiencing the same issue. I can pinpoint the start of the problem to 11:00 CET. No changes, network issues, updates on my client at that time. So something probably changed on the server side.

**Reply by duncanScalion** [reply]
same issue, same exact starting time as @PetBom noted. A lot more users from spain with xdrip webfollower has raised problem in fb groups (most of all DT1 parents as me :P)

just solving webfollower bridge problems will be perfect, as just december 09 was working

No changes in phones, apps, nightscout or anything related. After a few research tried @Navid200 solution (posted in FB xdrip group) https://navid200.github.io/xDrip/docs/Libre3_Follow.html but it doesn't solve the issue

**Reply by syska** [reply]
Many users around the world. It also started for me at 11 CET.

I found this: https://github.com/gillesvs/librelink/issues/39 since I also use this integration, but haven't tested there solution yet.

**Reply by Havanna22** [reply]
Many, many users affected. They just don't find this thread.

**Reply by Havanna22** [reply]
A comment: I am also using Gluroo which also gets the values from LibreLink org. app and that is working fine. Must be using some other technique to get the data.
Gluroo also shows the BG value in the notification field at the top all the time.
But Gluroo has bad alarm functions, can't set levels for example.

Long shot, but could XDrip+ get the BG values from Gluroo's notifications, i.e. using Gluroo as companion app?

**Comment by Navid200** [comment]
Would it help if you could run both the official linkup app and xDrip on the same phone and have the readings transfer from the linkup app to xDrip?
If yes, we may be able to add it as a companion app option.

**Reply by Havanna22** [reply]
Yes, that would be help

**Reply by duncanScalion** [reply]
that will be perfect. The Occam's razor of solutions

**Reply by Navid200** [reply]
Just to be clear this will require you to have two apps installed on the same phone, the official LibreLinkup app and xDrip.
And this requires the Linkup app to create notifications one for each reading.

Does it do that?
Would this be useful?

**Reply by duncanScalion** [reply]
that is perfect (at least) for me. And nothing messing with sensor/BT/NFC and official librelink in master phone to solve this issue.

**Reply by Navid200** [reply]
Can you install the Librelinkup app and confirm that it produces notifications for each reading?

**Reply by duncanScalion** [reply]
if you guide me in process... I have xDrip+ dfcbe80-2024.09.17 and librelinkup in same device, yes

**Reply by Navid200** [reply]
Can you tell me if librelinkup generates a notification for each reading?

**Reply by duncanScalion** [reply]
oh, I may have misunderstood you before, I was wondering inside Xdrip logged as webfollower...

as plain librelinkup user, it doesn't seem to generate any notification unless you have that app opened. And there is no widget or similar. only having librelinkup opened. 

Alarms may be considered as notification, but only at high/low triggered values

**Reply by Navid200** [reply]
Does the app have any setting that would make it create a notification for each reading?

**Reply by duncanScalion** [reply]
no. Just high alarm/low alarm and missed readings. Each one with user-definable values in a preset range

Seemed so nice to be true... tomorrow maybe some more experienced users with android coding experience may enlighten my comments. But as plain user can't give you more info.

**Reply by Havanna22** [reply]
LibreLinkUp, just like the Libre 3 app, just have very basic functionality. No widgets, one must open the app to see a reading.
As written above, no notifications that a regular user can see or configure - if there are any, they are in the background hidden.
Personally, I would have no problem having the LibreLinkUp (or Libre 3 app) open/active in the background all the time (but of course not with the interface shown) if that is required to get the readings easily into XDrip+.
(Have asked Abbott for a widget and other updates but of course no improvements from their side in years.)

**Reply by duncanScalion** [reply]
@Navid200, there is "Gluroo" android app suggested in other reply. Seems to work with notifications, hope this helps.

Edit: It has a ¿5-minute? updating icon(only cgm value) on android notifications bar and lock screen. In documentation is specifically written - can share bolus&meals with Nightscout but not cgm- . That point may be relevant, but as long as was written yesterday, this app notification seems to have the info needed to feed cgm to xdrip

**Comment by syska** [comment]
Issue here is from another app that also get the issue today: https://github.com/gillesvs/librelink/issues/39 - I think it's related ... 

I'm using xDrip to follow myself, so I can get the data to my Smart Watch. Also for the alarms, etc.

**Reply by Havanna22** [reply]
> Issue here is from another app that also get the issue today: [gillesvs/librelink#39](https://github.com/gillesvs/librelink/issues/39) - I think it's related ...
> 
> I'm using xDrip to follow myself, so I can get the data to my Smart Watch. Also for the alarms, etc.

Same here, use XDrip+ for same reasons

**Comment by HoLi1969** [comment]
Hello, I understood the solution is easy as this problem always happened when version of librelinkup changed. But someone needs to make changes in the config code as described in this thread. I am not a programmer so my hope is that this would come with a new nightly. Or did I understand the topic wrong?
https://github.com/JohanDegraeve/xdripswift/commit/13e4a1e7418364fbdb68e5bd54cac888fd2eb9b8

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by PetBom** [reply]
No, I think you understood it perfectly. Looking at the error response we are getting I would say that xDrip needs to support the API changes that is illustrated by @JohanDegraeve in 

1. LIbreLink version passed to the api needs to bea at least 4.12.0
2. The version key needs to be passed in sha256 (hashing algorithm) format. From looking at the code it seems to be passed as uint8 

@PhilTU  @gui-dos 

My guess is that Abbot updated the api, orretired earlier versions of the api yesterday, at least in some countries.

**Comment by duncanScalion** [comment]
there is nothing existing like a "patched librelinkup" (separate from xdrip app) that can/could generate those notifications that xdrip need?

 searching inside GitHub arise with some results, don't know how reusable that code could be

**Reply by Havanna22** [reply]
See below about Gluroo. Working with the LinkUp also after update and it generates notifications

**Reply by Havanna22** [reply]
![Screenshot_20241211_083718_One UI Home](https://github.com/user-attachments/assets/37e92a79-32ff-4d56-b20d-e010060ba83b)

**Comment by duncanScalion** [comment]
gluroo works fine in Android! (1st try connection, waiting to see how It works in longer time)

Hope this app can be used as source to xdrip...maybe this Will be solved setting a source, maybe existing. Going to check this as user

**Reply by Havanna22** [reply]
Yes, would be great if it could be added and working if selecting Companion App (for example) as Hardware Source in XDrip
But can't see any existing option that will get the values from Gluroo.

**Comment by HoLi1969** [comment]
And is there a possibility to help on this change? I am just a user as most here.

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________
